### PR TITLE
core/shamir: fix EMS vs MS (wrong base)

### LINKED ISSUE #400: Shamir Backups not behaving as expected
Using my new Trezor T (firmware 2.1.3) and beta-wallet.trezor.io I wiped and used "create a wallet with Shamir Backup".

-----

1st attempt : 
I did the backup, chose 3of5 and recorded the 20 words fragments

I recorded the ypub and xpub in the Basic tab, then in Advanced > Check recovery seed.

I entered the fragments #5, #3, #4.

> X Warning
> The entered recovery shares are valid but do not match what is currently in the device

I then wiped the Trezor T in beta-wallet and then "Recover your wallet" > 20 words.

Entering fragments #5, #3, #4 the ypub and xpub were different to those shown in beta-wallet before the wipe.

I then wiped again, and restored #1, #2, #3 with the same ypub/xpub as the #5, #3, #4 restoration.

-----
2nd attempt:

I wiped again, and in beta-wallet "Create a wallet with Shamir Backup" and recorded the 3of5 20-word-shares.

I recorded the ypub and xpub in beta-wallet

Then wiped Trezor

Then recovered fragments #3, #4, #5

beta-wallet assigned this restore a completely different device ID URL (and as i didn't forget the pre-wipe device) it created another trezor in the sidebar with a different ID, ypub and xpub.

---

I have not yet tried restoring a `trezorctl reset-device -t 256 -b shamir` seed yet to see if the bug is the same.

### PR DESCRIPTION
**Pull request by tsusanka**
We've decided the entropy is the Encrypted Master Secret, so that is what is stored in the storage, but in the generate_mnemonics function we need to have an EMS as argument instead of MS.

Fixes #400

### CODE REVIEW

**Comment by tsusanka** [comment]
@matejcik will have a look on the device test

File: core/src/trezor/crypto/slip39.py [hunk 1]
```diff
@@ -485,15 +485,15 @@ def generate_random_identifier() -> int:
 
 
 def generate_single_group_mnemonics_from_data(
-    master_secret: bytes,
+    encrypted_master_secret: bytes,
     identifier: int,
     threshold: int,
     count: int,
     passphrase: bytes = b"",
```
Comment (andrewkozlik) [inline core/src/trezor/crypto/slip39.py]: Remove the passphrase parameter. It no longer makes sense in this function.

File: core/src/trezor/crypto/slip39.py [hunk 2]
```diff
@@ -503,7 +503,7 @@ def generate_single_group_mnemonics_from_data(
 
 
 def generate_mnemonics_from_data(
```
Comment (andrewkozlik) [inline core/src/trezor/crypto/slip39.py]: The passphrase parameter is not used in the function. Please remove it.

File: core/src/apps/management/reset_device.py [hunk 3]
```diff
@@ -108,7 +111,10 @@
 
     # generate the mnemonics
     mnemonics = slip39.generate_single_group_mnemonics_from_data(
-        secret, storage.device.get_slip39_identifier(), threshold, shares_count
+        encrypted_master_secret,
+        storage.device.get_slip39_identifier(),
+        threshold,
+        shares_count,
```
Comment (andrewkozlik) [inline core/src/apps/management/reset_device.py]: Get the iteration exponent from the storage.

**Comment by matejcik** [comment]
closing in favor of #403

**Pull request closed without merging**